### ISSUE TITLE
PageIterator does not work on empty collections

### ISSUE THREAD

**Issue by ChristianBoehlke** [microsoftgraph/msgraph-sdk-javascript#129]
Hello,

We are using this library and the [`PageIterator`](https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/dev/src/tasks/PageIterator.ts) to iterate over multiple pages that are returned from `GET /v1.0/me/calendarView/delta`.

If the calendar is empty, the response looks like this:
```
/v1.0/me/calendarView/delta?startDateTime=[omitted]&endDateTime=[omitted]
{
  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#Collection(event)",
  "@odata.nextLink": "https: //graph.microsoft.com/v1.0/me/calendarview/delta?$skiptoken=[omitted]",
  "value": []
}
```

As you can see, `values` is empty but the response does not contain a `deltaLink` but a `nextLink`. Only on the second page (the second request) the response contains a `deltaLink`:

```
/v1.0/me/calendarview/delta?$skiptoken=[omitted]
{
  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#Collection(event)",
  "@odata.deltaLink": "https://graph.microsoft.com/v1.0/me/calendarview/delta?$deltatoken=[omitted]",
  "value": []
}
```

Unfortunately, the `PageIterator` stops iterating if the response has no values and it does not fetch the second page:
https://github.com/microsoftgraph/msgraph-sdk-javascript/blob/bed02c8eba445e0089a84fd2cb9d28da022ebfff/src/tasks/PageIterator.ts#L86-L88

Is there another way to make the `PageIterator` work with empty collections?

Would it make more sense to keep fetching pages until the first `deltaLink` is returned? I'm happy to contribute a PR for that.

**Comment by muthurathinam** [comment]
@ChristianBoehlke Thanks for reporting this issue. Yes its a valid to keep fetching until the `deltaLink` is returned in this case. It will be awesome if you contribute with a PR for this.

**Comment by muthurathinam** [comment]
Fixed in v1.5.2
